asyncio_mqtt/client.py

```diff
@@ -490,7 +530,7 @@ def _on_connect(
         client: mqtt.Client,
         userdata: Any,
         flags: Dict[str, int],
-        rc: int,
+        rc: int | mqtt.ReasonCodes,
```

Comment:
Nice touch of Python 3.10. :)) For the sake of backward-compatibility, let's use `Union` for now. With time, we can get rid of this legacy stuff (but not now).